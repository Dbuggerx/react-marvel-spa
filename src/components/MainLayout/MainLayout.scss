@import '../scss/variables';

.main-layout {
  display: grid;
  grid-template-columns: 1fr minmax(80px, 20vw);
  grid-template-rows: 40px 1fr 40px;
  grid-template-areas:
    'main-layout-search main-layout-info'
    'main-layout-list main-layout-info'
    'main-layout-pagination main-layout-info';
  height: 100%;
  width: 100%;
  background: rgba(map-get($colors, primary-bg), 0.85);
  color: map-get($colors, primary-fg);
  max-height: inherit;
  padding: 15px;
  box-sizing: border-box;
  position: relative;

  &__search {
    grid-area: main-layout-search;
  }

  &__list {
    grid-area: main-layout-list;
  }

  &__pagination {
    grid-area: main-layout-pagination;
    display: flex;
    flex-flow: row nowrap;
    align-content: center;
    justify-content: center;
    align-items: center;
  }

  &__info {
    grid-area: main-layout-info;
    margin-left: 15px;
  }

  @supports (display: grid) {
    gap: 15px;

    &__info {
      margin-left: 0;
    }

    @media #{map-get($breakpoints, medium)} {
      grid-template-columns: 1fr minmax(200px, 15vw);
    }

    @media #{map-get($breakpoints, small)} {
      grid-template-areas:
        'main-layout-search'
        'main-layout-list'
        'main-layout-pagination'
        'main-layout-info';

      grid-template-columns: 1fr;
      grid-template-rows: 40px 1fr 40px min-content;
    }
  }
}
