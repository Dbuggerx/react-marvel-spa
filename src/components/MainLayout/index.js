// @flow

import React from 'react';
import type { StatelessFunctionalComponent, Element } from 'react';
import './MainLayout.scss';

type Props = {
    search: Element<*>,
    list: Element<*>,
    info: Element<*>,
    pagination: ?*,
};

/**
 * Wraps application sections with specific "slots"
 */
const MainLayout: StatelessFunctionalComponent<Props> = (props: Props) => (
    <div className="main-layout">
        <div className="main-layout__search">{props.search}</div>
        <div className="main-layout__list" tabIndex="-1">
            {props.list}
        </div>
        <div className="main-layout__pagination">{props.pagination}</div>
        <div className="main-layout__info">{props.info}</div>
    </div>
);

MainLayout.displayName = 'MainLayout';

export default MainLayout;
