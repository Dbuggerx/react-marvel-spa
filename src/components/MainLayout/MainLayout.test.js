import React from 'react';
import { shallow } from 'enzyme';
import MainLayout from './index';

describe('MainLayout', () => {
    test('renders properly', () => {
        const wrapper = shallow(<MainLayout
            search="Search"
            list="List"
            pagination="Pagination"
            info="Information"
        />);
        expect(wrapper).toMatchSnapshot();
    });
});
