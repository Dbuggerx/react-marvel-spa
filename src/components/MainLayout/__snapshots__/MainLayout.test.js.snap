// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout renders properly 1`] = `
<div
  className="main-layout"
>
  <div
    className="main-layout__search"
  >
    Search
  </div>
  <div
    className="main-layout__list"
    tabIndex="-1"
  >
    List
  </div>
  <div
    className="main-layout__pagination"
  >
    Pagination
  </div>
  <div
    className="main-layout__info"
  >
    Information
  </div>
</div>
`;
